### PR TITLE
feat: Configs

### PR DESCRIPTION
**Pull request by inca**

### CODE REVIEW

File: src/main/util.ts [hunk 1]
```diff
@@ -45,3 +48,25 @@ export interface EntityList<T> {
...
+        const ownMetadata: T[] = Reflect.getOwnMetadata(key, target) || [];
+        result = ownMetadata.concat(result);
+        proto = Object.getPrototypeOf(proto);
+    }
+    return result;
+}
+
+export function getBindingsMap(container: Container):
+    Map<interfaces.ServiceIdentifier<any>, interfaces.Binding<any>[]> {
+    return (container as any)._bindingDictionary._map;
```
Comment (kyungeunni) [inline src/main/util.ts:71]: interesting to see to type inferring to `any`, but I assume type definition doesn't include `_bindingDictionary`

File: src/main/config.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    abstract resolve(key: string): string | null;
+
+    protected getOrNull<T>(
+        key: string,
+        parse: (str: string) => T | null,
+        defaultValue?: string | T,
+    ): T | null {
+        let str = this.resolve(key);
+        if (str == null) {
+            str = String(defaultValue);
```
Comment (kyungeunni) [inline src/main/config.ts:96]: if defaultValue is `undefined`, would it parse `'undefined'` if the `parse` argument is `parseString`?

**Review by kyungeunni — APPROVED** [review]
Nice to see how powerful the Reflect metadata is. Awesome work! 🌟

File: src/main/util.ts (hunk 1 above)
Comment (inca) [reply src/main/util.ts:71]: Yep, unfortunately they don't expose internals, so this is my way of accessing their `private` stuff

File: src/main/config.ts (hunk 2 above)
Comment (inca) [reply src/main/config.ts]: Fixed this one, thanks 🦅 👁️

**Review by TobiaszCudnik — APPROVED** [review]
LGTM, nice to have per-class configs, the naming convention is controversial to say at least.

**Pull request merged**